### ISSUE TITLE
New cop to enforce caller/callee method defs ordering

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#8236]
## Is your feature request related to a problem? Please describe.

One of the chapters of 'Clean Code' book  goes into a ton of detail on how R. Martin suggests ordering code. He suggests that reading code should work kind of like reading a newspaper article: the high-level details come first (at the top) and you get more detail as you read down. He says, "If one function calls another, they should be vertically close, and the caller should be above the callee, if at all possible." An excerpt from that chapter is [here](https://hashnode.com/post/clean-code-book-summary-part-4-formatting-cize2dwtw000gfh53qa72p7zu).

I have always implicitly and unknowingly adhered to those rules even before reading that book. So I would like to enforce those rules. Some people prefer callees before callers, so this can be configurable.

## Describe the solution you'd like

I would like to have a cop that tries to reorder method definitions to keep: 
1) callers and callees as close as possible
2) callers before callees

Example:
```ruby
# bad
class Foo
  def do_something_else
  end

  def do_something
    do_something_else
  end
end

# good
class Foo
  def do_something
    do_something_else
  end

  def do_something_else
  end
end
```

If this seems useful to other people too, I will try to implement a PR.

Update: also found similar recommendation in thoughtbot style guide - https://github.com/thoughtbot/guides/tree/master/general#organization
